src/Dscp/Educator/Secret/Real/Types.hs

```diff
@@ -0,0 +1,52 @@
+module Dscp.Educator.Secret.Real.Types
+    ( EducatorSecretParams (..)
+    , EducatorSecret (..)
+    , EducatorSecretJson (..)
+    , KeyfileContent
+    ) where
+
+import Data.Aeson (FromJSON (..), ToJSON (..), object, withObject, (.:), (.=))
+import Test.QuickCheck (Arbitrary (..))
+
+import Dscp.Crypto (Encrypted (..), PassPhrase, SecretKey)
+import Dscp.Educator.Secret.Types (EducatorSecret (..))
+import Dscp.Util.Aeson (AsByteString (..), Base64, Versioned)
+
+-- | Contains all parameters required for manipulating with secret key.
+data EducatorSecretParams = EducatorSecretParams
+    { espPath       :: !(Maybe FilePath)
+      -- ^ Path to file with secret key.
+      -- If not specified, some default OS-dependent path is used.
+    , espGenNew     :: !Bool
+      -- ^ When 'True', file with secret key is expected to be
+      -- absent and will be generated from scratch.
+      -- When 'False', file should be present and it will be used.
+    , espPassphrase :: !PassPhrase
+      -- ^ Password from encrypted secret key stored on disk.
+    }
+
+---------------------------------------------------------------------
+-- Instances
+---------------------------------------------------------------------
+
+-- | Intermediate form of 'EducatorSecret' for JSON serialization.
+data EducatorSecretJson = EducatorSecretJson
+    { esjEncSecretKey :: Encrypted SecretKey
```

Comment:
Missed bang-pattern. Also, why do you need this intermediate form anyway?

Reply:
Secret key is kept encrypted in file, while in context we have unencrypted one, and it seemed improper to involve encryption into JSON serialization.